solvcon/solver.py

```diff
@@ -53,27 +55,21 @@
     BaseSolverExedata, BaseSolver, FakeBlockVtk, BlockSolver)
 
 
-class _MethodList(list):
-    """
-    A custom :py:class:`list` that provides a decorator for keeping names of
-    functions.
-
-    >>> mmnames = _MethodList()
-    >>> @mmnames.register
-    ... def func_of_a_name():
-    ...     pass
-    >>> mmnames
-    ['func_of_a_name']
-
-    This class is a private helper and should only be used in
-    :py:mod:`solvcon.solver`.
-    """
+class MeshSolverMeta(type):
```

Comment:
No longer use the custom list type (_MethodList).  Use this metaclass (MeshSolverMeta) to fill _MMNAMES.